core/meta/src/TClass.cxx

```diff
@@ -5464,29 +5464,24 @@ TClass *TClass::LoadClassCustom(const char *requestedname, Bool_t silent)
 
 void TClass::LoadClassInfo() const
 {
-   R__LOCKGUARD(gInterpreterMutex);
+   // Load class info if not already loaded and auto-parsing is not suspended
+   if (!fClassInfo && !gInterpreter->IsAutoParsingSuspended()) {
```

Comment:
> I cannot see how a pointer can be only partially set.

Re-read the 'raison d'etre' of atomics.  Without fClassInfo being an atomic, there *are* race condition where a thread is writing to a memory area at the same time another thread is reading this same are *and* only a partial copy of the writing (let's the first 4 out of 8 bytes) is seen by the reader.